### PR TITLE
Added assets search suggestions #232

### LINKED ISSUE #232: Search Functionailty (Make dicussions & discord visible)
## (WIP)
Part of #159 - Asset search
Related to #2 - Searchable bevy

Based on being active in the bevy discord help channel, some of the questions asked have already been discussed. In order to intrinsically prevent duplicates, we should use the website as a medium to redirect users for help.

Goal: 
Encourage discovery though:
- [Discussions](https://github.com/bevyengine/bevy/discussions)
- [Discord](https://discord.gg/bevy)
- Cheatbook (https://bevy-cheatbook.github.io)
in order to unify help in one place for new-comers and to promote bevy assets

Pros & Cons:
```diff
+  Not having the costs of hosting a database
-  Not as customizable as it relies on GitHub and crates.io APIs
```

Requires metadata tags from github and or crates.io specified in #230 

[Prototype](https://www.figma.com/file/CZ1svP4l70eMuPYXIOK9d3/bevy_design_sheet?node-id=612%3A58):

Where the URL looks something like `https://bevyengine.org/assets?q=hide+mouse`

And clicking on _Search for “hide mouse” in discussions_ links to: 
[`https://github.com/bevyengine/bevy/discussions?discussions_q=hide+mouse`](https://github.com/bevyengine/bevy/discussions?discussions_q=hide+mouse) and 
[`https://bevy-cheatbook.github.io/?search=hide%20mouse`](https://bevy-cheatbook.github.io/?search=hide%20mouse)
 for bevy-cheatbook

Group 88

### PR DESCRIPTION
**Pull request by d-bucur**
Resolves #232

- When searching, adds some further suggestions at the bottom of the page. These redirect to other community resources with the same search parameters filled in where possible (excluding Discord)
- Added small search delay when typing to prevent too many refreshes. delay is configurable

[demo video](https://github.com/bevyengine/bevy-website/assets/6576777/0a606869-05b3-40c9-9130-9ebf0f2f31f0)

### CODE REVIEW

File: templates/assets.html [hunk 1]
```diff
@@ -65,6 +65,17 @@ <h3 class="asset-subsection" id="{{ section.title | slugify }}">
 
         {% endfor %}
         {% endfor %}
+        <div id="suggestions-footer">
+            <div class="media-content">
+                Didn't find what you were looking for? How about searching our other community resources?
```
Comment (TrialDragon) [inline templates/assets.html:70]:
```suggestion
                Didn't find what you were looking for? See what our other community resources have on your search query.
```
I feel like it wasn't exactly obvious that the links would already contain the search query; my intuition was that they'd just take me to the site with no query. Should clarify wording imo.

**Review by TrialDragon — APPROVED** [review]
I have one suggestion for clarifying the wording a bit, but overall LGTM.

File: templates/assets.html (hunk 1 above)
Comment (alice-i-cecile) [reply templates/assets.html:70]:
```suggestion
                Didn't find what you were looking for? See what your search reveals in our docs and in other corners of the community.
```

**Review by BD103 — APPROVED** [review]
I'm not too familiar with the code, but it works well when testing it locally! I especially like the delay between updates, which is helpful when scanning results while you are typing.

**Comment by alice-i-cecile** [comment]
@d-bucur if you rebase this I'm happy to merge this in for you :) Otherwise I'll put it up for adoption after a week or so and someone else will finish this up for you. Sorry for letting this languish!

**Review by doup — COMMENTED** [review]
Really nice addition!

I found a couple of tiny issues. I think it's specially interesting to apply the _debounce_ function to improve UX.

I also noticed some weird behavior when navigating back: the search input keeps the search term but the list shows unfiltered… but looks like this doesn't happen on production… so maybe we shouldn't worry about this. Let's see what happens once this is synced with `main`. 🤷‍♂️

**Comment by d-bucur** [comment]
@alice-i-cecile No problem. Going to have a look at the change requests tomorrow

**Comment by BD103** [comment]
Closing in favor of #1075. Thanks for the initial effort!

**Pull request closed without merging**